vnext/Shared/Networking/OriginPolicyHttpFilter.cpp

```diff
@@ -0,0 +1,638 @@
+using winrt::Windows::Web::Http::HttpMethod;
+using winrt::Windows::Web::Http::HttpRequestMessage;
+using winrt::Windows::Web::Http::HttpResponseMessage;
+using winrt::Windows::Web::Http::Filters::IHttpFilter;
+using winrt::Windows::Web::Http::Headers::HttpMediaTypeHeaderValue;
+using winrt::Windows::Web::Http::Headers::HttpRequestHeaderCollection;
+
+namespace Microsoft::React::Networking {
+
+#pragma region OriginPolicyHttpFilter
+
+// https://fetch.spec.whatwg.org/#forbidden-method
+/*static*/ set<const wchar_t *> OriginPolicyHttpFilter::s_forbiddenMethods = {L"CONNECT", L"TRACE", L"TRACK"};
+
+/*static*/ set<const wchar_t *> OriginPolicyHttpFilter::s_simpleCorsMethods = {L"GET", L"HEAD", L"POST"};
+
+/*static*/ set<const wchar_t *> OriginPolicyHttpFilter::s_simpleCorsRequestHeaderNames = {
+    L"Accept",
+    L"Accept-Language",
+    L"Content-Language",
+    L"Content-Type",
+    L"DPR",
+    L"Downlink",
+    L"Save-Data",
+    L"Viewport-Width",
+    L"Width"};
+
+/*static*/ set<const wchar_t *> OriginPolicyHttpFilter::s_simpleCorsResponseHeaderNames =
+    {L"Cache-Control", L"Content-Language", L"Content-Type", L"Expires", L"Last-Modified", L"Pragma"};
+
+/*static*/ set<const wchar_t *> OriginPolicyHttpFilter::s_simpleCorsContentTypeValues = {
+    L"application/x-www-form-urlencoded",
+    L"multipart/form-data",
+    L"text/plain"};
+
+// https://fetch.spec.whatwg.org/#forbidden-header-name
+// Chromium still bans "User-Agent" due to https://crbug.com/571722 //TODO: Remove?
+/*static*/ set<const wchar_t *> OriginPolicyHttpFilter::s_corsForbiddenRequestHeaderNames = {
+    L"Accept-Charset",
+    L"Accept-Encoding",
+    L"Access-Control-Request-Headers",
+    L"Access-Control-Request-Method",
+    L"Connection",
+    L"Content-Length",
+    L"Cookie",
+    L"Cookie2",
+    L"Date",
+    L"DNT",
+    L"Expect",
+    L"Host",
+    L"Keep-Alive",
+    L"Origin",
+    L"Referer",
+    L"TE",
+    L"Trailer",
+    L"Transfer-Encoding",
+    L"Upgrade",
+    L"Via"};
+
+/*static*/ set<const wchar_t *> OriginPolicyHttpFilter::s_corsForbiddenRequestHeaderNamePrefixes = {L"Proxy-", L"Sec-"};
+
+/*static*/ Uri OriginPolicyHttpFilter::s_origin{nullptr};
+
+/*static*/ void OriginPolicyHttpFilter::SetStaticOrigin(const char *url) {
+  if (url)
+    s_origin = Uri{to_hstring(url)};
+}
+
+/*static*/ bool OriginPolicyHttpFilter::IsSameOrigin(Uri const &u1, Uri const &u2) noexcept {
+  return u1.SchemeName() == u2.SchemeName() && u1.Host() == u2.Host() && u1.Port() == u2.Port();
+}
+
+/*static*/ bool OriginPolicyHttpFilter::IsSimpleCorsRequest(HttpRequestMessage const &request) noexcept {
+  // Ensure header is in Simple CORS white list
+  for (const auto &header : request.Headers()) {
+    if (s_simpleCorsRequestHeaderNames.find(header.Key().c_str()) == s_simpleCorsRequestHeaderNames.cend())
+      return false;
+
+    // Ensure Content-Type value is in Simple CORS white list, if present
+    if (header.Key() == L"Content-Type") {
```

Comment:
>== [](http://example.com/codeflow?start=21&length=2)

In other code you used case insensitive comparisson. See your comment `Headers are generally case-insensitive`

Reply:
Good catch. I'll update accordingly.